### PR TITLE
fix: Ensure that passing values for option group with PostgreSQL does not cause errors

### LINKED ISSUE #465: create_db_option_group = true is avoiding the creation of a Postrgres instance
## Description

When `engine` is "postgres" and `create_db_option_group` is "true" instance creation will fail.

The problem is that the module will skip the option group creation for postgres engines, but it'll use the name provided for the option group anyways, lines 12 and 96 of `main.tf`:

```
12 | create_db_option_group = var.create_db_option_group && var.engine != "postgres"
...
96 | option_group_name      = local.option_group
```

I think this is due to an old AWS behaviour: https://github.com/terraform-aws-modules/terraform-aws-rds/issues/62

- [x] ✋ I have searched the open/closed issues and my issue is not listed.

## Versions

- Module version [Required]:

- Terraform version:
```
Terraform v1.3.6
on darwin_amd64
```
- Provider version(s):
```
+ provider registry.terraform.io/hashicorp/aws v4.48.0
+ provider registry.terraform.io/hashicorp/random v3.4.3
```

## Reproduction Code [Required]

```
terraform {
  required_providers {
    aws = {
      source = "hashicorp/aws"
    }
  }
  required_version = ">= 0.15"
}


provider "aws" {
  region  = "us-west-2"
}

module "bug-report" {
  source  = "terraform-aws-modules/rds/aws"
  version = "5.2.2"

  allocated_storage            = 10
  create_db_option_group       = true
  db_subnet_group_name         = "staging-database"
  engine                       = "postgres"
  engine_version               = "13.7"
  family                       = "postgres13"
  identifier                   = "bug-report"
  instance_class               = "db.t4g.small"
  major_engine_version         = "13"
  option_group_name            = "bug-report"
  option_group_use_name_prefix = false
  username                     = "notadmin"
}

```

Set these two inputs like this:
```
  create_db_option_group       = true
  engine                       = "postgres"
```

## Expected behavior

The RDS instance and option group is created correctly

## Actual behavior

The option group is skipped and the RDS instance cannot be created because of that.

### Terminal Output Screenshot(s)

```
module.bug-report.module.db_instance.aws_db_instance.this[0]: Creating...
╷
│ Error: creating RDS DB Instance (bug-report): OptionGroupNotFoundFault: Specified OptionGroupName: bug-report not found.
│ 	status code: 404, request id: b1dc3003-dfdb-4a82-9877-87e295905ac8
│
│   with module.bug-report.module.db_instance.aws_db_instance.this[0],
│   on .terraform/modules/bug-report/modules/db_instance/main.tf line 32, in resource "aws_db_instance" "this":
│   32: resource "aws_db_instance" "this" {
```

### PR DESCRIPTION
**Pull request by bryantbiggs**
## Description
- Ensure that passing values for option group with PostgreSQL does not cause errors

## Motivation and Context
- Resolves #465
- Resolves #466

## How Has This Been Tested?
- [ ] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [ ] I have tested and validated these changes using one or more of the provided `examples/*` projects

- [x] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -93,7 +93,7 @@ module "db_instance" {
   vpc_security_group_ids = var.vpc_security_group_ids
   db_subnet_group_name   = local.db_subnet_group_name
   parameter_group_name   = local.parameter_group_name_id
-  option_group_name      = local.option_group
+  option_group_name      = var.engine != "postgres" ? local.option_group : null
```
Comment (paolostancato) [inline main.tf:96]: Why would you null it when you can define an actual option group? (honest question)

Comment (bryantbiggs) [reply main.tf:96]: https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_WorkingWithOptionGroups.html

> PostgreSQL does not use options and option groups. PostgreSQL uses extensions and modules to provide additional features. For more information, see [Supported PostgreSQL extension versions](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/CHAP_PostgreSQL.html#PostgreSQL.Concepts.General.FeatureSupport.Extensions).

Comment (paolostancato) [reply main.tf:96]: Thanks!

File: .github/workflows/pre-commit.yml [hunk 2]
```diff
@@ -8,6 +8,7 @@ on:
 
 env:
   TERRAFORM_DOCS_VERSION: v0.16.0
+  TFLINT_VERSION: v0.44.1
```
Comment (bryantbiggs) [inline .github/workflows/pre-commit.yml:11]: Using a version avoids a few GitHub API calls which is currently causing issues with some workflows. Looks like they might have changed some of the rates, not sure why this is now an issue but just using a version seems to resole this

**Review by antonbabenko — APPROVED** [review]
LGTM since it fixes the issue.

**Comment by antonbabenko** [comment]
This PR is included in version 5.2.3 :tada:

**Pull request merged**